lib/flutter_calendar_carousel.dart

```diff
@@ -313,85 +413,85 @@ class _CalendarState extends State<CalendarCarousel> {
+                            Center(
+                              child: DefaultTextStyle(
                                 style: (_localeDate.dateSymbols.WEEKENDRANGE.contains((index - 1 + firstDayOfWeek) % 7)) &&
                                         !isSelectedDay &&
                                         !isToday
-                                    ? widget.weekendTextStyle
+                                    ? widget.defaultWeekendTextStyle
                                     : isToday
-                                        ? widget.todayTextStyle
-                                        : textStyle,
-                                maxLines: 1,
+                                        ? widget.defaultTodayTextStyle
+                                        : defaultTextStyle,
+                                child: Text(
+                                  '${now.day}',
+                                  style: (_localeDate.dateSymbols.WEEKENDRANGE.contains((index - 1 + firstDayOfWeek) % 7)) &&
+                                          !isSelectedDay &&
+                                          !isToday
+                                      ? widget.weekendTextStyle
+                                      : isToday
+                                          ? widget.todayTextStyle
+                                          : textStyle,
+                                  maxLines: 1,
+                                ),
                               ),
                             ),
-                          ),
-                          widget.markedDatesMap != null
-                              ? Container(
-                                  height: double.infinity,
-                                  padding: EdgeInsets.only(bottom: 4.0),
-                                  child: Row(
-                                    crossAxisAlignment: CrossAxisAlignment.end,
-                                    mainAxisSize: MainAxisSize.max,
-                                    mainAxisAlignment: MainAxisAlignment.center,
-                                    children: _renderMarkedMap(now),
-                                  ),
-                                )
-                              : _renderMarked(now),
-                        ],
+                          ],
+                        ),
                       ),
-                    ),
+                      widget.markedDates == null
+                          ? Flex(
+                                  direction: widget.markedDatesDirection,
+                                  crossAxisAlignment: CrossAxisAlignment.end,
+                                  mainAxisSize: MainAxisSize.max,
+                                  mainAxisAlignment: MainAxisAlignment.center,
+                                  children: _renderMarkedMap(now),
+                                )
+                          : _renderMarked(now),
+                    ])
                   );
```

Comment:
![image](https://user-images.githubusercontent.com/27461460/50042399-915e7000-00a5-11e9-9f37-0a0971d809e2.png)

Currently, I am facing `overflowed` problem as image above with your code. Could you check this out? I think it is because you removed resenter out of `Stack`.

Reply:
Hum... It's not entirely because of that. I've checked on the GridView flutter documentation and by default it constructs square cells. Therefore, the height is limited by the width of the cell. That's why you need with the current implementation to change `childAspectRatio` to something less than 1 to be able to show some elements. For example, mine is set to 0.6 because I need to display a lot of DateMark lines. https://github.com/flutter/flutter/issues/13777
Another way would be to change the behavior of the grid by manually creating one (and providing the geometric sizes). https://gist.github.com/aloisdeniel/81d6a5241ca9e2f82f17cf3c001444c5

Reply:
I didn't put it in the Stack because I didn't want to have the elements over the button. By having them over the button, you can have some design bugs:
* Mark dates hiding the day cell
* The Mark Dates not showing clearly when their background is the same color as the flat button background.
* The flat button shape (circle) doesn't allow too much space in it

Reply:
So is this design you are looking into?
![image](https://user-images.githubusercontent.com/27461460/50043729-c88c4b80-00bc-11e9-8d4a-3de94c9fe82d.png)

I think it isn't what others want because those usually stays inside the `square` or `circle`.
![image](https://user-images.githubusercontent.com/27461460/50043736-ece82800-00bc-11e9-84ca-ea71290a43b7.png)

Above is most understood design for the calendar.

Reply:
No problem to modify this. However the example from https://github.com/wix/react-native-calendars separate marking (dot and squares) and periods (lines) by having them at different places. Should we do the same? Having dots and squares inside and the lines under?

Reply:
Yeah. It would be best to follow that experience.

Reply:
I will look into the gap problem. I can't make them into the stack as it was before though. By doing that I get a BoxConstraints exception due to `height: double.infinity` on the Container above `Row` of dates marks. If I remove the attribute, the date marks are not bottom-centered.
I tried to use a `Positioned` widget instead of a Container but it was too close from the number of the day (just right under the number, almost over it).
I might have two solutions for this, I will try each one and push another commit when I have time :)

Reply:
@akuma06 Yeah. I really appreciate what you are doing. It doesn't have to be perfect so just `PR` whenever you want to communicate. Also, I hope you to rebase this if you can since some changes been done in last `PR`.